uplc-builtins.md

```diff
@@ -24,5 +24,14 @@ module UPLC-BUILTINS
   rule <k> (builtin (_::PolyBuiltinName #Or chooseData) #as BN) ~> Force => < builtin BN .List #numArgs(BN) > ... </k>
        <env> _ => .Map </env>
 
+  rule #typeCheck(ListItem(< con T _ >)          L, T:TypeConstant TS:TypeSignature) => #typeCheck(L, TS)
```

Comment:
Yes, if the arguments are exhausted and properly type checked then the base case will apply. And if none of these type checks pass or it is somehow "over applied" then it will fail:
https://github.com/runtimeverification/plutus-core-semantics/blob/87cb7eacb421cd60b0eaffc9e0d1580472054f8f/uplc-builtins.md?plain=1#L33-L34

Reply:
Yes, but it is not suppose to fail if the arguments have the right type but not all of them were passed.